enderio-conduits/src/main/java/crazypants/enderio/conduits/gui/InventoryUpgrades.java

```diff
@@ -129,25 +122,11 @@ public boolean isItemValidForSlot(int slot, @Nonnull ItemStack stack, IConduit c
     return false;
   }
 
-  // TODO make a way to do this based on the conduit
   private boolean isFilterUpgradeAccepted(@Nonnull ItemStack stack, IConduit con, boolean isInput) {
```

Comment:
Please make both either methods or inline in the switch. I don't care which one, but they should be handled the same way.